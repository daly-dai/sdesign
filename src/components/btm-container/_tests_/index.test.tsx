import { render } from '@testing-library/react';
import { Button } from 'antd';
import React from 'react';
import { describe, expect, it } from 'vitest';

import mountTest from '../../../tests/mountTest';
import SBtmContainer from '../index'; // 假设这是你的组件文件路径

describe('SBtmContainer', () => {
  mountTest(SBtmContainer);
  mountTest(() => <SBtmContainer />);

  it('should support custom style', () => {
    const { container } = render(
      <SBtmContainer style={{ background: '#333' }} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should support props full', () => {
    const { container } = render(<SBtmContainer full />);

    expect(container).toMatchSnapshot();
  });

  it('should render children correctly', () => {
    const { container } = render(
      <SBtmContainer>
        <Button type="primary">取消</Button>
        <Button type="primary">提交</Button>
      </SBtmContainer>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should support custom icon className', () => {
    const { container } = render(
      <SBtmContainer className="custom-SBtmContainer" />,
    );
    expect(container.querySelectorAll('.custom-SBtmContainer').length).toBe(1);
    expect(container).toMatchSnapshot();
  });

  it('fix renders {0} , 0 and {false}', () => {
    expect(
      render(<SBtmContainer>{0}</SBtmContainer>).container.firstChild,
    ).toMatchSnapshot();
    expect(
      render(<SBtmContainer>0</SBtmContainer>).container.firstChild,
    ).toMatchSnapshot();
    expect(
      render(<SBtmContainer>{false}</SBtmContainer>).container.firstChild,
    ).toMatchSnapshot();
  });

  it('should merge text if children using variable', () => {
    const wrapper = render(
      <SBtmContainer>
        {/* eslint-disable-next-line react/jsx-curly-brace-presence */}
        This {'is'} a test {1}
      </SBtmContainer>,
    );
    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it('should render width empty children', () => {
    const { container } = render(
      <SBtmContainer>
        <span />
        <span />
      </SBtmContainer>,
    );

    expect(
      container.querySelector<HTMLDivElement>('div.ant-space'),
      // @ts-ignore
    ).toHaveClass('ant-space-gap-row-small');
    expect(
      container.querySelector<HTMLDivElement>('div.ant-space'),
      // @ts-ignore
    ).toHaveClass('ant-space-gap-col-small');
  });

  // 你可以添加更多测试用例来覆盖不同的 props 和场景
});
