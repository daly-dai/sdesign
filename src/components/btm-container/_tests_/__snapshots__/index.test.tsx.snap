// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SBtmContainer > fix renders {0} , 0 and {false} 1`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      0
    </div>
  </div>
</div>
`;

exports[`SBtmContainer > fix renders {0} , 0 and {false} 2`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      0
    </div>
  </div>
</div>
`;

exports[`SBtmContainer > fix renders {0} , 0 and {false} 3`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  />
</div>
`;

exports[`SBtmContainer > should merge text if children using variable 1`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      This 
    </div>
    <div
      class="ant-space-item"
    >
      is
    </div>
    <div
      class="ant-space-item"
    >
       a test 
    </div>
    <div
      class="ant-space-item"
    >
      1
    </div>
  </div>
</div>
`;

exports[`SBtmContainer > should render children correctly 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  >
    <div
      class="ant-space css-dev-only-do-not-override-3rel02 ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
    >
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary"
          type="button"
        >
          <span>
            取 消
          </span>
        </button>
      </div>
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary"
          type="button"
        >
          <span>
            提 交
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SBtmContainer > should support custom icon className 1`] = `
<div>
  <div
    class="custom-SBtmContainer"
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  />
</div>
`;

exports[`SBtmContainer > should support custom style 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(51, 51, 51); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  />
</div>
`;

exports[`SBtmContainer > should support full 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100% - 32px);"
  />
</div>
`;

exports[`SBtmContainer > should support props full 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100% - 32px);"
  />
</div>
`;

exports[`SBtmContainer fixbug renders {0} , 0 and {false} 1`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-1r287do ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      0
    </div>
  </div>
</div>
`;

exports[`SBtmContainer fixbug renders {0} , 0 and {false} 2`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-1r287do ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      0
    </div>
  </div>
</div>
`;

exports[`SBtmContainer fixbug renders {0} , 0 and {false} 3`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-1r287do ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  />
</div>
`;

exports[`SBtmContainer should merge text if children using variable 1`] = `
<div
  style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
>
  <div
    class="ant-space css-dev-only-do-not-override-1r287do ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
  >
    <div
      class="ant-space-item"
    >
      This 
    </div>
    <div
      class="ant-space-item"
    >
      is
    </div>
    <div
      class="ant-space-item"
    >
       a test 
    </div>
    <div
      class="ant-space-item"
    >
      1
    </div>
  </div>
</div>
`;

exports[`SBtmContainer should render children correctly 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  >
    <div
      class="ant-space css-dev-only-do-not-override-1r287do ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
    >
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-primary"
          type="button"
        >
          <span>
            取 消
          </span>
        </button>
      </div>
      <div
        class="ant-space-item"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-primary"
          type="button"
        >
          <span>
            提 交
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SBtmContainer should support custom icon className 1`] = `
<div>
  <div
    class="custom-SBtmContainer"
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  />
</div>
`;

exports[`SBtmContainer should support custom style 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(51, 51, 51); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100vw - 264px);"
  />
</div>
`;

exports[`SBtmContainer should support full 1`] = `
<div>
  <div
    style="position: fixed; bottom: 16px; background: rgb(255, 255, 255); padding: 16px 24px; box-shadow: 0px -9px 7px 0px rgba(0,0,0,0.04); display: flex; justify-content: flex-end; z-index: 1000; border-radius: 8px; width: calc(100% - 32px);"
  />
</div>
`;
